File: src/engine/commands/tellcommand.cpp
#include "tellcommand.h"

#include "realm.h"
#include "util.h"


#define super Command

TellCommand::TellCommand(QObject *parent) :
    super(parent) {

    setDescription("Tell something to another online player (no need to be in the same area). Only "
                   "this player will hear you.\n"
                   "\n"
                   "Example: tell earl Hey Earl, where are you?");
}

TellCommand::~TellCommand() {
}

void TellCommand::execute(Player *player, const QString &command) {

    super::prepareExecute(player, command);

    if (!assertWordsLeft("Tell who?")) {
        return;
    }

    QString userName = Util::capitalize(takeWord());
    Player *other = Realm::instance()->getPlayer(userName);
    if (!other || !other->session()) {
        send("%1 is not online.", userName);
        return;
    }

    QString message = takeRest();

    player->tell(other, message);
}

File: data/commands/tellcommand.js
function TellCommand() {
    Command.call(this);

    this.setDescription("Tell something to another online player (no need to be in the same " +
                        "room). Only this player will hear you.\n" +
                        "\n" +
                        "Example: tell earl Hey Earl, where are you?");
}

TellCommand.prototype = new Command();
TellCommand.prototype.constructor = TellCommand;

TellCommand.prototype.execute = function(player, command) {

    this.prepareExecute(player, command);

    if (!this.assertWordsLeft("Tell who?")) {
        return;
    }

    var userName = this.takeWord().capitalized();
    var other = Realm.getPlayer(userName);
    if (!other || !other.isOnline()) {
        this.send("%1 is not online.", userName);
        return;
    }

    var message = this.takeRest();

    player.tell(other, message);

    other.lastTell = player;
};

CommandRegistry.registerCommand("tell", new TellCommand());
